src/GitHub.App/Services/GitClient.cs

```diff
@@ -136,6 +135,33 @@ public Task<Remote> GetHttpRemote(IRepository repo, string remote)
             });
         }
 
+        public async Task<string> ExtractFile(IRepository repository, string commitSha, string fileName)
+        {
+            var commit = repository.Lookup<Commit>(commitSha);
+            var blob = commit[fileName]?.Target as Blob;
+
+            var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempFileName = $"{Path.GetFileNameWithoutExtension(fileName)}@{commitSha}{Path.GetExtension(fileName)}";
+            var tempFile = Path.Combine(tempDir, tempFileName);
+
+            Directory.CreateDirectory(tempDir);
+
+            if (blob != null)
+            {
+                using (var source = blob.GetContentStream(new FilteringOptions(fileName)))
+                using (var destination = File.OpenWrite(tempFile))
```

Comment:
The scope on this block is a bit hard to read. Probably better to be clear about the scope with some indentation here, or really clear  with `{}` on both (you know, like the `{}` on `if` statements 😉)